stl/inc/compare

```diff
@@ -78,44 +64,50 @@ public:
     }
 
     _NODISCARD friend constexpr bool operator<=(const partial_ordering _Val, _Literal_zero) noexcept {
-        return _Val._Value <= 0 && _Val._Is_ordered();
+        // The stored value is either less (0xff), equivalent (0x00), greater (0x01), or unordered (0x80).
+        // Subtracting 1 produces either 0xfe, 0xff, 0x00, or 0x7f. The high bit is set in the result
+        // if and only if the initial value was less or equivalent, for which we want to return true.
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-1117991
+        return static_cast<signed char>(static_cast<unsigned char>(_Val._Value) - 1) < 0;
+#else // ^^^ no workaround / workaround vvv
+        auto _Tmp = static_cast<unsigned char>(_Val._Value);
+        --_Tmp;
+        return static_cast<bool>(_Tmp >> 7);
+#endif // TRANSITION, DevCom-1117991

```

Comment:
```suggestion
        // The stored value is either less (0xff), equivalent (0x00), greater (0x01), or unordered (0x80).
        // Subtracting from 0 produces either 0x01, 0x00, 0xff, or 0x80. The result is greater than or equal to 0
        // if and only if the initial value was less or equivalent, for which we want to return true.
        return static_cast<signed char>(0 - static_cast<unsigned int>(_Val._Value)) >= 0;
```

It seems that this form (equivalent to `0 <=> _Val >= 0`) is [not affected](https://godbolt.org/z/njqEdT) by DevCom-1117991.

Reply:
Thanks - in addition to being simpler and compiler-independent, the fact that we're already using this technique for `0 <=> _Val` makes it easier to reason about, and doesn't increase our exposure to compiler bugs. And thanks for validating the codegen! I've merged your suggestion and will port it to the MSVC-internal PR.